src/plugins/intel_gpu/src/graph/impls/ocl/generic_layer.cpp

```diff
@@ -27,18 +27,21 @@ struct generic_layer_impl : typed_primitive_impl<generic_layer> {
 
     generic_layer_impl(const generic_layer_impl& other)
     : _cl_kernel_data(other._cl_kernel_data)
-    , _kernels({})
+    , _kernel(nullptr)
     , _cached_kernel_id(other._cached_kernel_id) {
-        if (other._kernels.empty()) {
-            throw std::runtime_error("Can't copy generic_layer_impl node: kernels vector is empty");
-        }
-        _kernels.push_back(other._kernels.front()->clone());
+        OPENVINO_ASSERT(other._kernel, "[GPU] Can't copy generic_layer_impl node: kernel is nullptr");
+        _kernel = other._kernel->clone();
     }
 
-    generic_layer_impl(const generic_layer_node& arg)
-        : _cl_kernel_data(*arg.get_primitive()->generic_params.clKernel.get())
-        , _kernels()
-        , _cached_kernel_id() { }
+    generic_layer_impl(const kernel_impl_params& params)
+        : _cl_kernel_data()
+        , _kernel(nullptr)
+        , _cached_kernel_id() {
+        auto reorder_params = params.typed_desc<generic_layer>()->params;
+        auto casted_params = std::dynamic_pointer_cast<WeightsReorderParamsOCL>(reorder_params);
+        OPENVINO_ASSERT(casted_params, "[GPU] Invalid weights reorder parameters type for ", params.desc->id, " node");
+        _cl_kernel_data = *casted_params->get_cl_kernel();
```

Comment:
Umm, are we going to define & confine generic layer only for weight reorder? 
If so.. why do we need "generic" layer?

Reply:
Yes, actually it is already used only for weights reordering, but maybe we need to consider to rename it for clarity...